specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,705 @@
+  /// Remove listener for ErrorReceived event.
+  HRESULT remove_ErrorReceived([in] EventRegistrationToken token);
+  /// Set the D3D device this texture stream should use for creating shared
+  /// texture resources. When the RenderAdapterLUIDChanged event is raised you
+  /// should create a new D3D device using the RenderAdapterLUID property and
+  /// call SetD3DDevice with the new D3D device.
+  /// See the `CreateTextureStream` `d3dDevice` parameter for more details.
+  HRESULT SetD3DDevice([in] IUnknown* d3dDevice);
+  /// Event handler for receiving texture by Javascript.
+  /// `window.chrome.webview.registerTextureStream` call by Javascript will
+  /// request sending video frame to the host where it will filter requested
+  /// page's origin against allowed origins. If allowed, the Javascript will
+  /// send a video frame (web texture), through MediaStreamTrack insertable APIs,
+  /// MediaStreamTrackGenerator.
+  /// https://www.w3.org/TR/mediacapture-transform/.
+  /// WebTextureReceived event will be called only when it receives
+  /// a web texture. There is no start event for receiving web texture.
+  HRESULT add_WebTextureReceived(
+      [in] ICoreWebView2StagingTextureStreamWebTextureReceivedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for WebTextureReceived event.
+  HRESULT remove_WebTextureReceived([in] EventRegistrationToken token);
+  /// Event handler for stopping of the receiving texture stream.
+  /// It is expected that the host releases any holding handle/resource from
+  /// the WebTexture before an event handler returns. The event can be raised
+  /// when the JS calls `window.chrome.webview.unregisterTextureStream`.
+  /// JS can restart sending the stream with `window.chrome.webview.registerTextureStream`
+  /// after stream is stopped. If the stream is started again, `add_WebTextureReceived`
+  /// will be called.
+  HRESULT add_WebTextureStreamStopped(
+      [in] ICoreWebView2StagingTextureStreamWebTextureStreamStoppedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for WebTextureStreamStopped event.
+  HRESULT remove_WebTextureStreamStopped([in] EventRegistrationToken token);
+}
+/// texture that the host writes to so that the Renderer
+/// will render on it.
+[uuid(0836f09c-34bd-47bf-914a-99fb56ae2d07), object, pointer_default(unique)]
+interface ICoreWebView2StagingTexture : IUnknown {
+    /// Returns Windows NT handle to shared memory containing the texture.
+    /// The caller expected to open it with ID3D11Device1::OpenSharedResource1
+    /// and writes the incoming texture to it.
+    [propget] HRESULT Handle([out, retval] HANDLE* value);
+    /// Returns IUnknown type that could be query interface to IDXGIResource.
+    /// The caller can write incoming texture to it.
+    [propget] HRESULT Resource([out, retval] IUnknown** value);
+    /// Sets timestamp of presenting texture.

```

Comment:
```suggestion
    /// The time in microseconds of the texture since the start of the video. The timestamp is the video capture time of this texture in microseconds since the start of the video stream. The value does not have to be an exact capture time, but it should be larger than previous frame's timestamps. WebView2 will ignore a video frame if the timestamp is equal or less than the previous frame.
    /// This value will be exposed to script as the [`VideoFrame.timestamp` property](https://developer.mozilla.org/en-US/docs/Web/API/VideoFrame/timestamp).
```